test/e2e/volume_test.go

```diff
@@ -0,0 +1,321 @@
+			Ref: v1beta1.TargetRef{
+				APIVersion: "v1",
+				Kind:       apis.KindPersistentVolumeClaim,
+				Name:       bpvc.Name,
+			},
+			VolumeMounts: []core.VolumeMount{
+				{
+					Name:      framework.TestSourceDataVolumeName,
+					MountPath: framework.TestSourceDataMountPath,
+				},
+			},
+		}
+		restoreSession.Spec.Rules = []v1beta1.Rule{
+			{
+				Paths: []string{
+					framework.TestSourceDataMountPath,
+				},
+			},
+		}
+		restoreSession.Spec.Task.Name = restoreTask.Name
+
+	})
+	AfterEach(func() {
+		err = f.DeleteSecret(cred.ObjectMeta)
+		Expect(err).NotTo(HaveOccurred())
+		err = framework.WaitUntilSecretDeleted(f.KubeClient, cred.ObjectMeta)
+		Expect(err).NotTo(HaveOccurred())
+	})
+	var (
+		testPVCBackup = func() {
+			By("Creating New PV and PVC")
+			err = f.CreatePersistentVolume(bpv)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = f.CreatePersistentVolumeClaim(bpvc)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Create Pod and Generate sample Data")
+			err = f.CreatePod(pod)
+			Expect(err).NotTo(HaveOccurred())
+			err = util.WaitUntilPodRunning(f.KubeClient, pod.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Reading sample data from /source/data directory inside pod")
+			sampleData, err = f.ReadSampleDataFromFromWorkload(pod.ObjectMeta, apis.KindPersistentVolumeClaim)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating storage Secret " + cred.Name)
+			err = f.CreateSecret(cred)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating new repository")
+			err = f.CreateRepository(repo)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating BackupConfiguration" + backupCfg.Name)
+			err = f.CreateBackupConfiguration(backupCfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for BackupSession")
+			f.EventuallyBackupSessionCreated(backupCfg.ObjectMeta).Should(BeTrue())
+			bs, err := f.GetBackupSession(backupCfg.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Check for succeeded BackupSession")
+			f.EventuallyBackupSessionPhase(bs.ObjectMeta).Should(Equal(v1beta1.BackupSessionSucceeded))
+
+			By("Delete BackupConfiguration")
+			err = f.DeleteBackupConfiguration(backupCfg)
+			err = framework.WaitUntilBackupConfigurationDeleted(f.StashClient, backupCfg.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Remove sample data from PVC")
+			err = f.CleanupSampleDataFromWorkload(pod.ObjectMeta, apis.KindPersistentVolumeClaim)
+			Expect(err).NotTo(HaveOccurred())
+			err = util.WaitUntilPodRunning(f.KubeClient, pod.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+		}
+	)
+	Context("General Backup && Restore for PVC Volume", func() {
+		BeforeEach(func() {
+			bpv = f.GetPersistentVolume()
+			bpvc = f.GetPersistentVolumeClaim()
+			fmt.Println(bpvc.Name)
+
+			updateStatusFunc = f.UpdateStatusFunction()
+			backupFunc = f.BackupFunction()
+			restoreFunc = f.RestoreFunction()
+
+			err = f.CreateFunction(updateStatusFunc)
```

Comment:
Move creation of `Function` and `Task` in `BeforeEach` of the Context/Describe and reuse them in other cases.